Upstream-Status: Backport [https://github.com/openssl/openssl/commit/6e42e3ff9cde43830555549fdafa2a8b37b9485b.patch, https://mta.openssl.org/pipermail/openssl-dev/2016-March/005758.html]
From 6e42e3ff9cde43830555549fdafa2a8b37b9485b Mon Sep 17 00:00:00 2001
From: Andy Polyakov <appro@openssl.org>
Date: Fri, 4 Mar 2016 11:32:26 +0100
Subject: [PATCH] perlasm/x86_64-xlate.pl: handle binary constants early.

Not all assemblers of "gas" flavour handle binary constants, e.g.
seasoned MacOS Xcode doesn't, so give them a hand.

Reviewed-by: Rich Salz <rsalz@openssl.org>
Reviewed-by: Viktor Dukhovni <viktor@openssl.org>
---
 crypto/perlasm/x86_64-xlate.pl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/crypto/perlasm/x86_64-xlate.pl b/crypto/perlasm/x86_64-xlate.pl
index 0a023fb..80ab17f 100755
--- ./crypto/perlasm/x86_64-xlate.pl
+++ ./crypto/perlasm/x86_64-xlate.pl
@@ -195,6 +195,7 @@
     sub out {
     	my $self = shift;
 
+	$self->{value} =~ s/\b(0b[0-1]+)/oct($1)/eig;
 	if ($gas) {
 	    # Solaris /usr/ccs/bin/as can't handle multiplications
 	    # in $self->{value}
@@ -205,7 +206,6 @@
 	    }
 	    sprintf "\$%s",$self->{value};
 	} else {
-	    $self->{value} =~ s/(0b[0-1]+)/oct($1)/eig;
 	    $self->{value} =~ s/0x([0-9a-f]+)/0$1h/ig if ($masm);
 	    sprintf "%s",$self->{value};
 	}
